elasticsearch/example/src/test/java/example/springdata/elasticsearch/conference/ElasticsearchOperationsTest.java

```diff
@@ -44,24 +44,30 @@
  * @author Oliver Gierke
  * @author Christoph Strobl
  * @author Prakhar Gupta
+ * @author Peter-Josef Meisch
  */
-@SpringBootTest(classes = ApplicationConfiguration.class)
+@SpringBootTest(classes = { ApplicationConfiguration.class, ElasticsearchOperationsTest.TestConfiguration.class })
 @Testcontainers
 class ElasticsearchOperationsTest {
 
 	private static final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd");
-
 	@Container //
 	private static ElasticsearchContainer container = new ElasticsearchContainer(
-			DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch:7.17.2")) //
-					.withPassword("foobar") //
-					.withReuse(true);
-
-	@DynamicPropertySource
-	static void setProperties(DynamicPropertyRegistry registry) {
-		registry.add("spring.elasticsearch.uris", () -> "http://" + container.getHttpHostAddress());
-		registry.add("spring.elasticsearch.username", () -> "elastic");
-		registry.add("spring.elasticsearch.password", () -> "foobar");
+			DockerImageName.parse("docker.elastic.co/elasticsearch/elasticsearch:8.7.0")) //
+			.withPassword("foobar");
```

Comment:
Is there a reason for removing `withReuse(true)`?

Reply:
Also the tests don't work for me. They get stuck with
```
13:26:51.980 [main] INFO  org.testcontainers.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
13:26:52.093 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
```

Reply:
> Is there a reason for removing `withReuse(true)`?

Testcontainers output showed me  a hint that this function is called although not being supported by the environment therefore it's ignored, so I removed it.

Tests were running on my side, can that be a problem with downloading the image? I'll try again later this evening when at home

Reply:
When adding the reuse I get this message:
> WARN  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Reuse was requested but the environment does not support the reuse of containers
To enable reuse of containers, you must set 'testcontainers.reuse.enable=true' in a file located at /Users/peter/.testcontainers.properties

# first test run

Besides that the tests run through on my machine (I have the images already pulled):

```shell
╭─peter@gandalf.local ~/Entwicklung/spring/spring-data-examples/elasticsearch  ‹main›
╰─➤  ../mvnw test                                                                                                                                                                                                                                                                                                                                                     1 ↵
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] Spring Data Elasticsearch - Examples                               [pom]
[INFO] Spring Data Elasticsearch - High Level REST Client Example         [jar]
[INFO] Spring Data Elasticsearch - Reactive Example                       [jar]
[INFO]
[INFO] --< org.springframework.data.examples:spring-data-elasticsearch-examples >--
[INFO] Building Spring Data Elasticsearch - Examples 2.0.0.BUILD-SNAPSHOT [1/3]
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --< org.springframework.data.examples:spring-data-elasticsearch-example >--
[INFO] Building Spring Data Elasticsearch - High Level REST Client Example 2.0.0.BUILD-SNAPSHOT [2/3]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ spring-data-elasticsearch-example ---
[INFO] Copying 0 resource from src/main/resources to target/classes
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO]
[INFO] --- maven-compiler-plugin:3.11.0:compile (default-compile) @ spring-data-elasticsearch-example ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:3.3.1:testResources (default-testResources) @ spring-data-elasticsearch-example ---
[INFO] skip non existing resourceDirectory /Users/peter/Entwicklung/spring/spring-data-examples/elasticsearch/example/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.11.0:testCompile (default-testCompile) @ spring-data-elasticsearch-example ---
[INFO] Changes detected - recompiling the module! :source
[INFO] Compiling 1 source file with javac [debug release 17] to target/test-classes
[INFO]
[INFO] --- maven-surefire-plugin:3.0.0:test (default-test) @ spring-data-elasticsearch-example ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running example.springdata.elasticsearch.conference.ElasticsearchOperationsTest
20:58:15.658 [main] INFO  org.testcontainers.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
20:58:16.099 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
20:58:16.631 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
20:58:16.632 [main] INFO  org.testcontainers.DockerClientFactory - Docker host IP address is localhost
20:58:16.660 [main] INFO  org.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.21
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 5943 MB
20:58:16.704 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Creating container for image: testcontainers/ryuk:0.4.0
20:58:17.556 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Container testcontainers/ryuk:0.4.0 is starting: f8a3b75bdfad465ad2feda2b34bd93e62ca25fcccba89e3f2dafdfeacca4236c
20:58:18.618 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Container testcontainers/ryuk:0.4.0 started in PT1.947127S
20:58:18.667 [main] INFO  org.testcontainers.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
20:58:18.667 [main] INFO  org.testcontainers.DockerClientFactory - Checking the system...
20:58:18.668 [main] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
20:58:18.671 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Creating container for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0
20:58:19.333 [main] INFO  org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for docker.elastic.co
20:58:19.334 [main] WARN  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Reuse was requested but the environment does not support the reuse of containers
To enable reuse of containers, you must set 'testcontainers.reuse.enable=true' in a file located at /Users/peter/.testcontainers.properties
20:58:19.563 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 is starting: 4f5681caa64990804e03e2b97c837f2a3055824d9ecc8d9aa7a89847b4eca68e
20:58:59.590 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 started in PT40.921767S

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::            (v3.1.0-RC1)

20:59:00.368 [main] INFO  example.springdata.elasticsearch.conference.ElasticsearchOperationsTest - Starting ElasticsearchOperationsTest using Java 17.0.6 with PID 70844 (started by peter in /Users/peter/Entwicklung/spring/spring-data-examples/elasticsearch/example)
20:59:00.369 [main] INFO  example.springdata.elasticsearch.conference.ElasticsearchOperationsTest - No active profile set, falling back to 1 default profile: "default"
20:59:00.969 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Bootstrapping Spring Data Elasticsearch repositories in DEFAULT mode.
20:59:01.024 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 48 ms. Found 1 Elasticsearch repository interfaces.
20:59:01.035 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Bootstrapping Spring Data Reactive Elasticsearch repositories in DEFAULT mode.
20:59:01.043 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 7 ms. Found 0 Reactive Elasticsearch repository interfaces.
20:59:01.512 [main] WARN  org.springframework.data.elasticsearch.core.mapping.SimpleElasticsearchPersistentProperty - Unsupported type 'class java.lang.String' for date property 'date'.
20:59:04.605 [main] INFO  example.springdata.elasticsearch.conference.ElasticsearchOperationsTest - Started ElasticsearchOperationsTest in 4.663 seconds (process running for 51.334)
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 52.732 s - in example.springdata.elasticsearch.conference.ElasticsearchOperationsTest
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO]
[INFO] --< org.springframework.data.examples:spring-data-elasticsearch-reactive-example >--
[INFO] Building Spring Data Elasticsearch - Reactive Example 2.0.0.BUILD-SNAPSHOT [3/3]
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:3.3.1:resources (default-resources) @ spring-data-elasticsearch-reactive-example ---
[INFO] Copying 0 resource from src/main/resources to target/classes
[INFO] Copying 1 resource from src/main/resources to target/classes
[INFO]
[INFO] --- maven-compiler-plugin:3.11.0:compile (default-compile) @ spring-data-elasticsearch-reactive-example ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:3.3.1:testResources (default-testResources) @ spring-data-elasticsearch-reactive-example ---
[INFO] skip non existing resourceDirectory /Users/peter/Entwicklung/spring/spring-data-examples/elasticsearch/reactive/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.11.0:testCompile (default-testCompile) @ spring-data-elasticsearch-reactive-example ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-surefire-plugin:3.0.0:test (default-test) @ spring-data-elasticsearch-reactive-example ---
[INFO] Using auto detected provider org.apache.maven.surefire.junitplatform.JUnitPlatformProvider
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running example.springdata.elasticsearch.conference.ReactiveElasticsearchRepositoryTest
20:59:09.003 [main] INFO  org.testcontainers.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
20:59:09.250 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
20:59:09.688 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
20:59:09.690 [main] INFO  org.testcontainers.DockerClientFactory - Docker host IP address is localhost
20:59:09.718 [main] INFO  org.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.21
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 5943 MB
20:59:09.759 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Creating container for image: testcontainers/ryuk:0.4.0
20:59:10.467 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Container testcontainers/ryuk:0.4.0 is starting: d8c7181410513cb9f64c5c1b138b430359af8d6f66a02219447d561ab85edd15
20:59:11.064 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Container testcontainers/ryuk:0.4.0 started in PT1.333584S
20:59:11.099 [main] INFO  org.testcontainers.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
20:59:11.101 [main] INFO  org.testcontainers.DockerClientFactory - Checking the system...
20:59:11.106 [main] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
20:59:11.107 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Creating container for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0
20:59:11.740 [main] INFO  org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for docker.elastic.co
20:59:12.062 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 is starting: d92a4b4d56a52459dd350cfde1cdaf7e89a4365e43a3168f3ac31a3f3dc66140
20:59:48.193 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 started in PT37.087001S

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::            (v3.1.0-RC1)

20:59:48.869 [main] INFO  example.springdata.elasticsearch.conference.ReactiveElasticsearchRepositoryTest - Starting ReactiveElasticsearchRepositoryTest using Java 17.0.6 with PID 70984 (started by peter in /Users/peter/Entwicklung/spring/spring-data-examples/elasticsearch/reactive)
20:59:48.870 [main] INFO  example.springdata.elasticsearch.conference.ReactiveElasticsearchRepositoryTest - No active profile set, falling back to 1 default profile: "default"
20:59:49.769 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Bootstrapping Spring Data Elasticsearch repositories in DEFAULT mode.
20:59:50.031 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 254 ms. Found 0 Elasticsearch repository interfaces.
20:59:50.040 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Bootstrapping Spring Data Reactive Elasticsearch repositories in DEFAULT mode.
20:59:50.065 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 24 ms. Found 1 Reactive Elasticsearch repository interfaces.
20:59:51.059 [main] WARN  org.springframework.data.elasticsearch.core.mapping.SimpleElasticsearchPersistentProperty - Unsupported type 'class java.lang.String' for date property 'date'.
20:59:53.556 [main] INFO  example.springdata.elasticsearch.conference.ReactiveElasticsearchRepositoryTest - Started ReactiveElasticsearchRepositoryTest in 5.185 seconds (process running for 46.088)
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 47.682 s - in example.springdata.elasticsearch.conference.ReactiveElasticsearchRepositoryTest
[INFO] Running example.springdata.elasticsearch.conference.ReactiveElasticsearchOperationsTest
20:59:55.889 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Creating container for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0
20:59:56.061 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 is starting: dcbf231563c9ae10eb4fbbabafc895d64038e153835b598d5d32c4974280f4c9
21:00:32.603 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 started in PT36.714556S

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::            (v3.1.0-RC1)

21:00:32.953 [main] INFO  example.springdata.elasticsearch.conference.ReactiveElasticsearchOperationsTest - Starting ReactiveElasticsearchOperationsTest using Java 17.0.6 with PID 70984 (started by peter in /Users/peter/Entwicklung/spring/spring-data-examples/elasticsearch/reactive)
21:00:32.953 [main] INFO  example.springdata.elasticsearch.conference.ReactiveElasticsearchOperationsTest - No active profile set, falling back to 1 default profile: "default"
21:00:33.184 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Bootstrapping Spring Data Elasticsearch repositories in DEFAULT mode.
21:00:33.192 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 7 ms. Found 0 Elasticsearch repository interfaces.
21:00:33.194 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Bootstrapping Spring Data Reactive Elasticsearch repositories in DEFAULT mode.
21:00:33.205 [main] INFO  org.springframework.data.repository.config.RepositoryConfigurationDelegate - Finished Spring Data repository scanning in 10 ms. Found 1 Reactive Elasticsearch repository interfaces.
21:00:33.334 [main] WARN  org.springframework.data.elasticsearch.core.mapping.SimpleElasticsearchPersistentProperty - Unsupported type 'class java.lang.String' for date property 'date'.
21:00:35.840 [main] INFO  example.springdata.elasticsearch.conference.ReactiveElasticsearchOperationsTest - Started ReactiveElasticsearchOperationsTest in 2.911 seconds (process running for 88.374)
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 41.321 s - in example.springdata.elasticsearch.conference.ReactiveElasticsearchOperationsTest
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for Spring Data Elasticsearch - Examples 2.0.0.BUILD-SNAPSHOT:
[INFO]
[INFO] Spring Data Elasticsearch - Examples ............... SUCCESS [  0.002 s]
[INFO] Spring Data Elasticsearch - High Level REST Client Example SUCCESS [01:02 min]
[INFO] Spring Data Elasticsearch - Reactive Example ....... SUCCESS [01:30 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:33 min
[INFO] Finished at: 2023-05-03T21:00:37+02:00
[INFO] ------------------------------------------------------------------------
╭─peter@gandalf.local ~/Entwicklung/spring/spring-data-examples/elasticsearch  ‹main*›
╰─➤
```
# second test run

I then deleted the docker images and re-ran the tests, downloading had some hickups but succeeded and so did the tests (only showing the output until the first test starts):

```shell
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running example.springdata.elasticsearch.conference.ElasticsearchOperationsTest
21:07:41.361 [main] INFO  org.testcontainers.utility.ImageNameSubstitutor - Image name substitution will be performed by: DefaultImageNameSubstitutor (composite of 'ConfigurationFileImageNameSubstitutor' and 'PrefixingImageNameSubstitutor')
21:07:41.649 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Loaded org.testcontainers.dockerclient.UnixSocketClientProviderStrategy from ~/.testcontainers.properties, will try it first
21:07:42.297 [main] INFO  org.testcontainers.dockerclient.DockerClientProviderStrategy - Found Docker environment with local Unix socket (unix:///var/run/docker.sock)
21:07:42.299 [main] INFO  org.testcontainers.DockerClientFactory - Docker host IP address is localhost
21:07:42.326 [main] INFO  org.testcontainers.DockerClientFactory - Connected to docker:
  Server Version: 20.10.21
  API Version: 1.41
  Operating System: Docker Desktop
  Total Memory: 5943 MB
21:07:42.396 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling docker image: testcontainers/ryuk:0.4.0. Please be patient; this may take some time but only needs to be done once.
21:07:44.897 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Starting to pull image
21:07:44.993 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
21:07:45.887 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  2 pending,  1 downloaded,  0 extracted, (334 KB/? MB)
21:07:46.417 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  1 pending,  2 downloaded,  0 extracted, (5 MB/? MB)
21:07:46.455 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  0 pending,  3 downloaded,  0 extracted, (5 MB/5 MB)
21:07:46.674 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  0 pending,  3 downloaded,  1 extracted, (5 MB/5 MB)
21:07:46.823 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  0 pending,  3 downloaded,  2 extracted, (5 MB/5 MB)
21:07:47.061 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pulling image layers:  0 pending,  3 downloaded,  3 extracted, (5 MB/5 MB)
21:07:47.129 [docker-java-stream--1627952871] INFO  tc.testcontainers/ryuk:0.4.0 - Pull complete. 3 layers, pulled in 2s (downloaded 5 MB at 2 MB/s)
21:07:47.173 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Creating container for image: testcontainers/ryuk:0.4.0
21:07:47.354 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Container testcontainers/ryuk:0.4.0 is starting: 107bb3379f35ed2e29f85669504d44425de5e12de2c9326ebc7b9332b9e124ae
21:07:48.240 [main] INFO  tc.testcontainers/ryuk:0.4.0 - Container testcontainers/ryuk:0.4.0 started in PT5.903617S
21:07:48.255 [main] INFO  org.testcontainers.utility.RyukResourceReaper - Ryuk started - will monitor and terminate Testcontainers containers on JVM exit
21:07:48.255 [main] INFO  org.testcontainers.DockerClientFactory - Checking the system...
21:07:48.255 [main] INFO  org.testcontainers.DockerClientFactory - ✔︎ Docker server version should be at least 1.6.0
21:07:48.260 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling docker image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0. Please be patient; this may take some time but only needs to be done once.
21:07:48.856 [main] INFO  org.testcontainers.utility.RegistryAuthLocator - Credential helper/store (docker-credential-desktop) does not have credentials for docker.elastic.co
21:07:59.496 [docker-java-stream--1373245132] ERROR com.github.dockerjava.api.async.ResultCallbackTemplate - Error during callback
com.github.dockerjava.api.exception.InternalServerErrorException: Status 500: {"message":"Get \"https://docker.elastic.co/v2/\": net/http: TLS handshake timeout"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247) ~[testcontainers-1.18.0.jar:1.18.0]
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.lambda$executeAndStream$1(DefaultInvocationBuilder.java:269) ~[testcontainers-1.18.0.jar:1.18.0]
	at java.lang.Thread.run(Thread.java:833) [?:?]
21:07:59.512 [main] WARN  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Retrying pull for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0 (108s remaining)
21:08:18.635 [docker-java-stream--1373245132] ERROR com.github.dockerjava.api.async.ResultCallbackTemplate - Error during callback
com.github.dockerjava.api.exception.InternalServerErrorException: Status 500: {"message":"Head \"https://docker.elastic.co/v2/elasticsearch/elasticsearch/manifests/8.7.0\": Get \"https://docker-auth.elastic.co/auth?scope=repository%3Aelasticsearch%2Felasticsearch%3Apull&service=token-service\": net/http: TLS handshake timeout"}

	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247) ~[testcontainers-1.18.0.jar:1.18.0]
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.lambda$executeAndStream$1(DefaultInvocationBuilder.java:269) ~[testcontainers-1.18.0.jar:1.18.0]
	at java.lang.Thread.run(Thread.java:833) [?:?]
21:08:18.636 [main] WARN  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Retrying pull for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0 (89s remaining)
21:08:28.662 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Starting to pull image
21:08:28.663 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
21:08:59.147 [testcontainers-pull-watchdog-2] ERROR tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Docker image pull has not made progress in 30s - aborting pull
21:08:59.148 [main] WARN  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Retrying pull for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0 (49s remaining)
21:09:05.088 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Starting to pull image
21:09:05.089 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending,  0 downloaded,  0 extracted, (0 bytes/0 bytes)
21:10:04.854 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  9 pending,  1 downloaded,  0 extracted, (0 bytes/? MB)
21:10:05.276 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  8 pending,  2 downloaded,  0 extracted, (39 bytes/? MB)
21:10:12.424 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  7 pending,  3 downloaded,  0 extracted, (70 MB/? MB)
21:10:13.663 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  6 pending,  4 downloaded,  0 extracted, (80 MB/? MB)
21:10:14.487 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  5 pending,  5 downloaded,  0 extracted, (90 MB/? MB)
21:10:15.445 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  4 pending,  6 downloaded,  0 extracted, (101 MB/? MB)
21:10:16.067 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  3 pending,  7 downloaded,  0 extracted, (107 MB/? MB)
21:10:16.206 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  2 pending,  8 downloaded,  0 extracted, (109 MB/? MB)
21:10:16.664 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  1 pending,  9 downloaded,  0 extracted, (114 MB/? MB)
21:10:25.903 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  1 pending,  9 downloaded,  1 extracted, (222 MB/? MB)
21:10:27.560 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  1 pending,  9 downloaded,  2 extracted, (240 MB/? MB)
21:10:27.983 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  1 pending,  9 downloaded,  3 extracted, (245 MB/? MB)
21:10:28.615 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  1 pending,  9 downloaded,  4 extracted, (252 MB/? MB)
21:11:05.061 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded,  4 extracted, (668 MB/669 MB)
21:11:28.501 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded,  5 extracted, (668 MB/669 MB)
21:11:28.843 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded,  6 extracted, (668 MB/669 MB)
21:11:29.044 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded,  7 extracted, (668 MB/669 MB)
21:11:29.160 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded,  8 extracted, (669 MB/669 MB)
21:11:29.249 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded,  9 extracted, (669 MB/669 MB)
21:11:29.427 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pulling image layers:  0 pending, 10 downloaded, 10 extracted, (669 MB/669 MB)
21:11:29.484 [docker-java-stream--1373245132] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Pull complete. 10 layers, pulled in 144s (downloaded 669 MB at 4 MB/s)
21:11:29.503 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Creating container for image: docker.elastic.co/elasticsearch/elasticsearch:8.7.0
21:11:29.508 [main] WARN  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Reuse was requested but the environment does not support the reuse of containers
To enable reuse of containers, you must set 'testcontainers.reuse.enable=true' in a file located at /Users/peter/.testcontainers.properties
21:11:36.838 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 is starting: 00f9d0bcf4fd5d972747d51fc81ec69da82dcbb43b54e217ccade69cdd0fdac8
21:12:14.149 [main] INFO  tc.docker.elastic.co/elasticsearch/elasticsearch:8.7.0 - Container docker.elastic.co/elasticsearch/elasticsearch:8.7.0 started in PT4M25.893276S

  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::            (v3.1.0-RC1)
```